src/array_api_stubs/_draft/data_type_functions.py

```diff
@@ -31,6 +43,8 @@ def astype(x: array, dtype: dtype, /, *, copy: bool = True) -> array:
         desired data type.
     copy: bool
         specifies whether to copy an array when the specified ``dtype`` matches the data type of the input array ``x``. If ``True``, a newly allocated array must always be returned. If ``False`` and the specified ``dtype`` matches the data type of the input array, the input array must be returned; otherwise, a newly allocated array must be returned. Default: ``True``.
+    device: Optional[device]
+        device on which to place the created array. Default: ``None``.
```

Comment:
```suggestion
        device on which to place the created array. If ``device`` is ``None``, the output array device must be inferred from ``x``. Default: ``None``.
```

Similar to `ones_like`, `empty_like`, and `full_like`, IMO, we should infer the output array device from the input array when the `device` is not explicitly specified.

The alternative would be to allocate on the "default" device, but that makes less sense to me given that we already infer other info from the input array (e.g., `shape`). So having consistency with the `*-like` functions seems to me to be most prudent.